packages/mui-material/src/Grid2/grid2Classes.ts

```diff
@@ -1,19 +1,40 @@
-import { GridClasses } from '@mui/system/Grid';
 import generateUtilityClasses from '@mui/utils/generateUtilityClasses';
 import generateUtilityClass from '@mui/utils/generateUtilityClass';
 
-export type Grid2ClassKey = keyof GridClasses;
-
-export function getGrid2UtilityClass(slot: string): string {
-  return generateUtilityClass('MuiGrid2', slot);
-}
-
 const SPACINGS = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10] as const;
 const DIRECTIONS = ['column-reverse', 'column', 'row-reverse', 'row'] as const;
 const WRAPS = ['nowrap', 'wrap-reverse', 'wrap'] as const;
 const GRID_SIZES = ['auto', true, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12] as const;
 
-const grid2Classes: GridClasses = generateUtilityClasses('MuiGrid2', [
+type Grid2DynamicClasses = {
```

Comment:
Let's export this as well. I have seen some TS compilation issues in the past if all the related types are not exported.
```suggestion
export type Grid2DynamicClasses = {
```